File: DFS/Cpp/11724.cpp
#include <iostream>
#include <vector>
using namespace std;
vector<int> v[1001];
bool check[1001];
void dfs(int s) {
    for (int i = 0; i < v[s].size(); i++) {
        int a = v[s][i];
        if (!check[a]) {
            check[a] = true;
            dfs(a);
        }
    }
}
int main() {
    int n, m;
    cin >> n >> m;
    for (int i = 0; i < m; i++) {
        int a, b;
        cin >> a >> b;
        v[a].push_back(b);
        v[b].push_back(a);
    }
    int count = 0;
    for (int i = 1; i <= n; i++) {
        if (!check[i]) {
            count++;
            check[i] = true;
            dfs(i);
        }
    }
    cout << count << '\n';
}

File: DFS/Python/11724.py
import sys
sys.setrecursionlimit(10000)


def dfs(u):
    check[u] = True
    for v in arr[u]:
        if not check[v]:
            dfs(v)


N, M = map(int, sys.stdin.readline().split())
arr = [[] for _ in range(N+1)]
check = [False] * (N+1)
count = 0
for _ in range(M):
    a, b = map(int, sys.stdin.readline().split())
    arr[a].append(b)
    arr[b].append(a)
for i in range(1, N+1):
    if not check[i]:
        dfs(i)
        count += 1
print(count)
